### DISCUSSION TITLE
How to debug IKE_AUTH request in detail?

### DISCUSSION THREAD

**Discussion by misch42** [strongswan/strongswan#2285] [General]
Hi,

I have a weird problem. I set up normal TS (localm remote) in my config, but my VPN partner tells me TS_UNACCEPT. In host log file, it seems that my gateway would ask for a 224.0.0.0/8 as my local TS. But this local_ts does not appear in list-conns. All fine here.

How can I debug the "generating IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr AUTH SA TSi TSr N(EAP_ONLY) iN(MSG_ID_SYN_SUP) ]" in detail?, especially the TSi and the TSr?

Thanks.

Michael.

**Comment by tobiasbrunner** [comment]
> I set up normal TS (localm remote) in my config

What does "normal TS" mean?

> it seems that my gateway would ask for a 224.0.0.0/8 as my local TS.

As your only TS? Or as additional TS?

> But this local_ts does not appear in list-conns.

Sounds like you might have some plugin loaded that modifies the traffic selectors. `224.0.0.0/8` is part of the IPv4 multicast address range (`224.0.0.0/4`). So maybe related to [forecast](https://docs.strongswan.org/docs/5.9/plugins/forecast.html)? However,  that plugin doesn't actually modify the traffic selectors. The [unity](https://docs.strongswan.org/docs/5.9/plugins/unity.html) plugin does modify them, but only for IKEv1. So no idea what the culprit could be in your case (I'm assuming you don't use the [addrblock](https://docs.strongswan.org/docs/5.9/plugins/addrblock.html) plugin, which is the only other plugin that modifies traffic selectors).

> How can I debug the "generating IKE_AUTH request 1 [ IDi N(INIT_CONTACT) IDr AUTH SA TSi TSr N(EAP_ONLY) iN(MSG_ID_SYN_SUP) ]" in detail?, especially the TSi and the TSr?

You see the traffic selectors (before modification through plugins) on level 2 in _cfg_. What's eventually getting sent is logged on level 3 in _enc_.

**Reply by misch42** [reply]
normal TS: "192.168.16.0/24"

Only TS: I don't know. That is not in the log files from the other side. The other side only complains about the 224.0.0.0/8 TS.

As far as I see, I do not use the forcast plugin. Is is enabled somehow by default?

I will try the level 2 and level 3 debugging. Will update this discussion.

**Reply by tobiasbrunner** [reply]
> Only TS: I don't know. That is not in the log files from the other side. The other side only complains about the 224.0.0.0/8 TS.

So they don't see what TS are actually received? Only the error? Could it just be an issue on the other end?

> As far as I see, I do not use the forcast plugin. Is is enabled somehow by default?

No, and you'd see it in the log or via `swanctl --stats`.

**Comment by misch42** [comment]
Thanks. Adding debug to the logging worked. I could prove that the the 224.0.0.0/8 was not suggested from our side.
